homeassistant/components/thermostat/proliphix.py

```diff
@@ -0,0 +1,206 @@
+    '4.5.6': 'LastUsageReset'
+}
+
+
+def _get_oid(name):
+    """Get OID by name/value."""
+    for oid, value in OIDS.items():
+        if value == name:
+            return oid
+    return None
+
+
+def _all_oids():
+    """Build a query string for all the OIDS we have."""
+    return "&".join([("OID" + x + "=") for x in OIDS.keys()])
+
+
+class PDP(object):
+    """PDP class for interacting with Proliphix thermostat.
+
+    OID states come back largely as enumerations, temp values come
+    back as an int which is in decidegrees.
+
+    The manual says don't make API calls more than once a minute for
+    prolonged periods of time, so we only refresh the data on update,
+    then fetch everything out of the cached data when needed.
+
+    """
+    def __init__(self, host, user, passwd):
+        self._host = host
+        self._user = user
+        self._passwd = passwd
+        self._data = {}
+
+    def update(self):
+        url = "http://%s/get" % self._host
+        data = _all_oids()
+        r = requests.post(url, auth=(self._user, self._passwd), data=data)
+        for line in r.text.split('&'):
+            if line:
+                oid, value = line.split('=')
+                const = OIDS.get(oid[3:])
+                if const:
+                    self._data[const] = value
+        logger.debug("PDP collected data %s" % self._data)
+
+    def _set(self, **kwargs):
+        data = {}
+        for key, value in kwargs.items():
+            oid = _get_oid(key)
+            if oid:
+                data["OID%s" % oid] = value
+        url = "http://%s/pdp" % self._host
+        form_data = urlencode(data)
+        form_data += "&submit=Submit"
+        requests.post(url, auth=(self._user, self._passwd), data=form_data)
+
+    @property
+    def cur_temp(self):
+        return float(self._data['AverageTemp']) / 10
+
+    @property
+    def setback_heat(self):
+        return float(self._data['SetbackHeat']) / 10
+
+    @setback_heat.setter
+    def setback_heat(self, val):
+        self._data['SetbackHeat'] = int(val * 10)
+        self._set(SetbackHeat=self._data['SetbackHeat'])
+
+    @property
+    def hvac_state(self):
+        return int(self._data['HvacState'])
+
+    @property
+    def name(self):
+        return self._data['DevName']
+
+    @property
+    def fan_state(self):
+        if self._data['FanState'] == "2":
+            return "On"
+        else:
+            return "Off"
+
+
+class ProliphixThermostat(ThermostatDevice):
+    """ Represents a Proliphix thermostat. """
+
+    def __init__(self, host, user, passwd):
+        self._pdp = PDP(host, user, passwd)
+        # initial data
+        self._pdp.update()
+        self._name = self._pdp.name
+
+    @property
+    def should_poll(self):
+        return True
+
+    def update(self):
+        self._pdp.update()
+
+    # change this so that we graph more often and collect the
+    # current_temperature when it changes.
+    @property
+    def state(self):
```

Comment:
Please do not do this.

I know there is a problem with the graphs but adding a hack inside one platform is not the solution.